geopandas/geoseries.py

```diff
@@ -53,6 +53,37 @@ def _geoseries_constructor_with_fallback(data=None, index=None, crs=None, **kwar
         return Series(data=data, index=index, **kwargs)
 
 
+def _geoseries_expanddim(data=None, index=None, crs=None, **kwargs):
+    from geopandas import GeoDataFrame
+
+    # DataFrame == pd.Series._constructor_expanddim
+    df = pd.DataFrame(data=data, index=index, **kwargs)
+    geo_col_name = None
+    if isinstance(data, GeoSeries):
+        # pandas default column name is 0, keep convention
+        geo_col_name = data.name if data.name is not None else 0
+
+    elif isinstance(data, dict) and len(data) == 1:
+        geo_col_name = list(data.keys())[0]
+    elif isinstance(data, BlockManager) and len(data.items) == 1:
+        geo_col_name = data.items[0]
+
+    if (df.dtypes == "geometry").sum() > 0:
+        df = GeoDataFrame(df)
```

Comment:
You can move this line inside the first `if` block, since the `set_geometry` in the `else` block will also ensure that `df` is a GeoDataFrame